echopype/convert/set_groups_base.py

```diff
@@ -162,30 +163,35 @@ def set_provenance(self) -> xr.Dataset:
         ds = ds.assign_attrs(prov_dict)
         return ds
 
+    @abc.abstractmethod
     def set_env(self) -> xr.Dataset:
         """Set the Environment group.
         """
-        pass
+        return xr.Dataset()
```

Comment:
These are a lot of instantiation for an abstract interface. Should probably avoid that for a base class, It would probably be better if we have a `raise NotImplementedError`.
```
@abc.abstractmethod
def set_env(self):
    """Set the Environment group.
    """
    raise NotImplementedError
```

Reply:
Ah right, thank you